### PR TITLE
124 perm

### LINKED ISSUE #124: Only authors can create/edit/delete questions on their events

### PR DESCRIPTION
**Pull request by ljones140**
closes #124

### CODE REVIEW

**Comment by Yorkshireman** [comment]
Could you confer with Ben about getting rid of the commented code in routes.rb for errors? Also, currently all errors are routed to a view called 'not found', even though it's always because of a 'not found' error. I think this is messy code that needs tidying up - would like to see that part of it pristine and 100% justifiable. Might just need to change the name of it to something more generic. Or may need more tinkering.

**Comment by ljones140** [comment]
@Yorkshireman I will speak to ben but if you look at my code the errors I have redirected to root path with a flash error ```redirect_to root_path, notice: "Sorry, but we were unable to serve your request."```
so this pull request is not part of this discussion

File: spec/features/questions_feature_spec.rb [hunk 1]
```diff
@@ -53,6 +54,13 @@
     expect(page).to have_content "1 error prohibited this question from being saved:"
   end
 
+  it 'disallows non owners to make event questions' do
+    click_on "Sign out"
+    sign_in_as(user_two)
+    visit event_path(event)
```
Comment (Yorkshireman) [inline spec/features/questions_feature_spec.rb]: This should probably be:  
`visit new_event_question_path(event)`  
Direct targets the proper path. The current line tests that the user can't visit someone else's event.

File: spec/features/questions_feature_spec.rb [hunk 2]
```diff
@@ -87,6 +95,15 @@
       expect(page).to have_selector(:link_or_button, "Back")
     end
 
+    it 'cannot be accessed expect by event author' do
```
Comment (Yorkshireman) [inline spec/features/questions_feature_spec.rb]: except

File: spec/features/questions_feature_spec.rb (hunk 1 above)
Comment (ljones140) [inline spec/features/questions_feature_spec.rb]: No this works

Comment (ljones140) [inline spec/features/questions_feature_spec.rb]: Another test could be added for visit ```new_event_question_path(event)``` yes but we also need to test this one

**Comment by Yorkshireman** [comment]
Good work on the controllers and testing. I think the error handling should be sorted out before merging.

Do we even need the custom error handling, now that we have proper redirects at controller level? If custom error handling is going to be introduced, I think it should be coded 100% properly - it's a bit messy at the moment.

**Comment by Yorkshireman** [comment]
Does the code you've done here interact with the custom error handling that Ben made?

**Comment by Yorkshireman** [comment]
If not, then we could merge it and deal with the error handling separately. If it does though, it might be better to separate the interaction here before merging?

**Comment by ljones140** [comment]
@Yorkshireman it doesn't have anything to do with the error handling.

File: spec/features/questions_feature_spec.rb (hunk 1 above)
Comment (Yorkshireman) [inline spec/features/questions_feature_spec.rb]: It works, as in the test passes, but visiting event_path(event) should be an Event Features test, testing that a user can't visit someone else's event. This test doesn't test that a user can't create a question on someone else's event.

To test that a user can't create a question on someone else's event, you need to test that they can't visit `new_event_question_path(event)` where `event` is someone else's event.

Comment (ljones140) [inline spec/features/questions_feature_spec.rb]: @Yorkshireman I see I can have a look tomorrow now

**Pull request merged**